code-challenges.test.js

```diff
@@ -14,43 +14,120 @@
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 // a) Create a test with expect statements using the variables provided.
+describe("codedMessage", function(){
+    it("takes in a string and returns a coded message", function(){
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
 
 // b) Create the function that makes the test pass.
 
+// create a function name codedmessage that takes in a string
+// make an array theat stores that coded letters
+// make a conditional statement inside a loop where a string will be added if the word includes a letter of the coded message to the new variable of string
+// return the coded message
+
+function codedMessage(string) {
+    let arr = ["4", "3", "1", "0"]
+    let newString = ""
+    for (let i = 0; i < string.length; i++) {
+        if (string.charAt(i).toLowerCase() === "a") {
+            newString += arr[0]
+        } else if (string.charAt(i).toLowerCase() === "e") {
+            newString += arr[1]
+        } else if (string.charAt(i).toLowerCase() === "i") {
+            newString += arr[2]
+        } else if (string.charAt(i).toLowerCase() === "o") {
+            newString += arr[3]
+        } else {
+            newString += string[i]
+        }
+    }
+    return newString
+}
```

Comment:
Your logic looks great!  Can you think of how you can refactor to use a higher-order function instead of a for loop?